### ISSUE TITLE
Simplify default build process

### ISSUE THREAD

**Issue by rethab** [actions/typescript-action#283]
Hi, from what I understand your build process basically looks like this:
- `npm run build` --> generates `lib/index.js`
- `npm run package` --> runs `ncc` which takes `lib/index.js` and generates `dist/index.js`

I think this could be simplified by:
- Remove `npm run build` action
- Change `"main"` in `package.json` to `src/index.ts` (this is what `ncc` takes as input)

This would mean the build process could be reduced to:
- `npm run package` --> runs `ncc` which now takes `src/index.ts` and generates `dist/index.js`

Or is there any particular reason you're using the "indirection" via `tsc`?
This may look like a small change to you, but note that this repository is copied everywhere so it would be nice to simplify it. And if that's not possible, it would be good to document why.

Thanks :)

**Comment by crazy-max** [comment]
@rethab Agree with that. Would also need to indicate `es2015` support:

```js
{
  "compilerOptions": {
    "target": "es2015",
    "moduleResolution": "node"
  }
}
```

**Comment by cdeutsch** [comment]
Just got confused by the lib/dist thing so +1 for simplifying. 

Would be even better if you could build/package on the fly so you didn't have to checkin the `dist` folder (which I've forgotten to do and spent hours trying to figure out what went wrong 🤦‍♂️ )

**Comment by rethab** [comment]
> Would be even better if you could build/package on the fly so you didn't have to checkin the `dist` folder (which I've forgotten to do and spent hours trying to figure out what went wrong 🤦‍♂️ )

How would you do that?

Note that forgetting to update the `dist` folder is a common problem. What you would typically do to mitigate this is to add a step to the workflow of your action that makes sure you didn't forget to update the `dist` folder. You can do this by running generating the `dist` folder as part of the workflow and then check whether anything changed. If something changed, you forgot to update the folder. Here's an example of how this could work in practice: https://github.com/actions/checkout/blob/main/.github/workflows/check-dist.yml

**Comment by cdeutsch** [comment]
> How would you do that?

I don't know and when I looked into it yesterday couldn't figure it out. I assume this is something GitHub would have to implement?

Thanks for the link to check-dist. I'll add that 👍

**Comment by rethab** [comment]
> I don't know and when I looked into it yesterday couldn't figure it out. I assume this is something GitHub would have to implement?

I think you want to have it packaged by the action author, because otherwise the action might behave differently based on what versions of tools are used when packaging it on the fly.

What they could do instead would be to allow publishing actions to a registry (like npmjs.org) and then make them usable like in a workflow like so:

```yaml
uses: npmjs://@actions/checkout
```

But this is kinda ugly and they started with the `dist` approach. Changing it now might lead to friction. Also, while having the packaged action in the `dist` folder is not very nice from an engineering perspective, it is very straightforward as less tools are involved.

**Comment by ncalteen** [comment]
👋🏻 Hello! Apologies for the delay on this issue. I am going through this repository and making some large updates. This will be resolved by #732
